projects/ngrx.io/content/guide/data/entity-metadata.md

```diff
@@ -227,6 +227,7 @@ If the property you want to add comes from `backend`, you will need some additio
 Create a new class `AdditionalPersistenceResultHandler` that `extends DefaultPersistenceResultHandler` and overwrite the [handleSuccess](https://github.com/ngrx/platform/blob/master/modules/data/src/dataservices/persistence-result-handler.service.ts) method, the purpose is to parse the data received from `DataService`, retrieve the additional property, and then save this to the `action.payload`. Here is an example.
 
 ```typescript
+@Injectable()
```

Comment:
Should we add `providedIn: 'root'`?

```suggestion
@Injectable({ providedIn: 'root' })
```

Reply:
No, it might be that the developer is using this service isolated in a lazy module! In this case you don't want it to be added to the application root. The developer should decide what to do with the service. Maybe we should add "providedIn" as an alternative and add a reference to the advantages and disadvantages of  providedIn: 'root'.

Reply:
I'm not clear on your reasoning. What do you mean by "isolation"?

Let me be clear about my thoughts on this:  if the service is a scalar singleton, it should almost always be `providedIn: 'root'`. By scalar singleton I mean that
* scalar: it is not multi-provided
* singleton: there will only ever be one instance of the service in the lifetime of the app.

It does not matter if the service is only used in a lazy loaded module. That is good conceptual scoping. But conceptual scoping is unrelated to `providedIn` which (in this case) is an implementation detail.

Of course there are exceptions. If this case is such an exception, we should add a code comment explaining why we're not using `providedIn` so that future readers don't try to add it.

Reply:
@wardbell
Lazy-Module:
This was not about lazy loading but about conceptual scoping as you described it. Depending on how (providedIn) and where (lazyModule) it is defined, the service (AdditionalPersistenceResultHandler) can be bundled in the main application or in the lazy-loaded file. Please correct me if I am wrong.

Here you describe how to do ngrx/data with lazy-loaded modules: https://ngrx.io/guide/data/entity-metadata#register-metadata

Isolated:
You're absolutely right about what you described being isolated! Exactly that's was what I meant but unfortunately I haven't described it very well. It was basically about good code organisation. I know the concept of Skalar and Singleton and know how it works in a NgModule.